staging/src/k8s.io/apiserver/pkg/storage/errors.go

```diff
@@ -82,30 +84,45 @@ func NewUnreachableError(key string, rv int64) *StorageError {
 
 func NewTimeoutError(key, msg string) *StorageError {
 	return &StorageError{
-		Code:               ErrCodeTimeout,
-		Key:                key,
-		AdditionalErrorMsg: msg,
+		Code: ErrCodeTimeout,
+		Key:  key,
+		err:  errors.New(msg),
 	}
 }
 
 func NewInvalidObjError(key, msg string) *StorageError {
 	return &StorageError{
-		Code:               ErrCodeInvalidObj,
-		Key:                key,
-		AdditionalErrorMsg: msg,
+		Code: ErrCodeInvalidObj,
+		Key:  key,
+		err:  errors.New(msg),
+	}
+}
+
+// NewCorruptObjError returns a new StorageError, it represents a corrupt object:
+// a) object data retrieved from the storage failed to transform with the given err.
+// b) the given object failed to decode with the given err
+func NewCorruptObjError(key string, err error) *StorageError {
+	return &StorageError{
+		Code: ErrCodeCorruptObj,
+		Key:  key,
+		err:  err,
 	}
 }
 
 type StorageError struct {
-	Code               int
-	Key                string
-	ResourceVersion    int64
-	AdditionalErrorMsg string
+	Code            int
+	Key             string
+	ResourceVersion int64
+
+	// inner error
+	err error
 }
 
+func (e *StorageError) Unwrap() error { return e.err }
+
 func (e *StorageError) Error() string {
-	return fmt.Sprintf("StorageError: %s, Code: %d, Key: %s, ResourceVersion: %d, AdditionalErrorMsg: %s",
-		errCodeToMessage[e.Code], e.Code, e.Key, e.ResourceVersion, e.AdditionalErrorMsg)
+	return fmt.Sprintf("StorageError: %s, Code: %d, Key: %s, ResourceVersion: %d, err: %v",
```

Comment:
This change seems capricious.  Why change `AdditionalErrorMsg` to `err`?

Reply:
sure, it stays in the message as is, I have a separate PR for this commit https://github.com/kubernetes/kubernetes/pull/128056